### ISSUE TITLE
Progress messages should go to stderr, not stdout

### ISSUE THREAD

**Issue by jidanni** [OSGeo/gdal#8468]
gdal2xyz.py has no way to turn off progress messages.
it needs a
"-q     Suppress progress monitor and other non-error output."
like gdal_create.

Also progress messages should go to stderr, not stdout! Same with gdal_grid.

**Comment by jratike80** [comment]
I am not a programmer but gdal2xyz.py is Python so maybe it cannot be directly compared with gdal_create.
About stderr/stdout I can say nothing.

**Comment by rouault** [comment]
> Also progress messages should go to stderr, not stdout!

this is quite disputed in https://unix.stackexchange.com/questions/331611/do-progress-reports-logging-information-belong-on-stderr-or-stdout . I'm not sure all GDAL utilities are consistent regarding this, but for now I wouldn't change that

**Comment by jidanni** [comment]
Well okay...  will just need elaborate filters to clean the progress messages out of standard output when piping to other programs.

**Comment by rouault** [comment]
> will just need elaborate filters to clean the progress messages out of standard output when piping to other programs.

hum, good point. Most GDAL programs have not both an interesting standard output that can be piped and a progress bar, but I see that "gdal2xyz.py input.tif /vsistdout/". But in that situation, a -q option is probably better indeed

**Comment by jidanni** [comment]
Yes. And about stdout vs. stderr.  That's how [K&R](https://en.wikipedia.org/wiki/K%26R) would do it.
